pkg/controller/job/job_controller.go

```diff
@@ -1448,7 +1458,7 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 			jm.enqueueSyncJobWithDelay(logger, job, remainingTime)
 			return 0, metrics.JobSyncActionPodsCreated, nil
 		}
-		diff := wantActive - active
+		diff := wantActive - active - terminating
```

Comment:
I think it will be easier to reason that the feature does not break anything if we had:
```
if hasPodReplacementPolicyFailed() {
  // Do not recreate terminating pods when podReplacementPolicy=Failed.
  diff -= int32(len(jobCtx.terminatingPods))
}
```
The there is also a natural place to have a comment.

Reply:
Updated this.